### DISCUSSION TITLE
[newbie] "Offline/watercooler" data sharing mode?

### DISCUSSION THREAD

**Discussion by akavel** [beakerbrowser/beaker#1686] [General]
Is there potentially some kind of a Secure-Scuttlebutt-like "offline sharing" mode for websites created with Beaker Browser?

**Comment by pfrazee** [comment]
Can you expand a little more on what "offline sharing" is? Is it a way to share content over the wifi but not over the Internet?

**Reply by akavel** [reply]
Hm, more or less. Actually I can think of a few possible complementary modes of operation here, but doing it over local wifi is definitively one of them. The other one(s) could be e.g. over a file (e.g. something like a [dead-drop](https://en.wikipedia.org/wiki/Dead_drop) USB pendrive), or bluetooth (maybe also via file?). The guys at Secure Scuttlebut seem to have this [cool video](https://scuttlebutt.nz/docs/introduction/) that I found helped me understand some of their ideas in this area.

**Reply by pfrazee** [reply]
Yeah I'm not sure we have that quite yet. For USB-sharing what you need is a way to export and import the data via a file. Blue-tooth is a question of adding it as a transport, which shouldn't be terribly difficult to add.

**Comment by georgebyron** [comment]
Hi, I'm also interested in this feature, so basically: 
can I share the content over the wifi but not over the Internet?
Just among friends that are part of the same network that is disconnected with the Internet?

So the basic flow to would be like this:
1. We're all connected to the same wifi network.
2. wifi network is not connected to the Internet
3. One of us provide link to her page on hyperdrive (the page hasn't been previously hosted on any other devices)
4. We click it and the page is displayed in the Beaker

thanks!

**Reply by matthewauld** [reply]
So, short answer is, yes- probably!
It has less to do with beaker than the underlying hyperswarm discovery methods. It currently tries to connect in two ways. The Distributed Hash Table (how you find remote sites) and  local multicast.  If you aren't connected to the internet you wont be able to use the DHT, but you can still discover local peers. Take a look at [Hyperswarm](https://github.com/hyperswarm) for more info. 

It also works on ad hoc mesh networks too (on Linux anyways, from my minimal mucking about with it last night), so you might not even need a wifi network eventually!

**Reply by pfrazee** [reply]
Yep this is accurate. And our meshnet support will improve when IPv6 support lands.

Having a way to broadcast sites over the wifi (Paul is sharing X sites) would also be cool. A little trickier since you gotta consider coffee shops and etc, but would love to see that.
